sycl/doc/SYCLRuntimeRFC.md

```diff
@@ -0,0 +1,97 @@
+scourse topic details
+
+- Category: [Runtimes](https://discourse.llvm.org/c/runtimes/64)
+- Title: "RFC: SYCL runtime upstreaming"
+- Tags: sycl
+
+## Overview
+
+Our SYCL runtime provides an implementation of the SYCL 2020 API specification.
+It's responsible for managing resources for enqueuing tasks to the offload
+device, tracking dependencies between them, and data movement between the host
+and devices. The SYCL runtime is device-agnostic and uses Unified Runtime
+(https://github.com/oneapi-src/unified-runtime) as an external dependency that
+serves as an interface layer between the SYCL runtime and device-specific
+backends. Unified Runtime has several adapters that bind to various backends in
+a similar fashion to libomptarget.
+
+This RFC contains a brief overview of the core functionality of the SYCL runtime
+and its major components. The main RFC for SYCL implementation upstreaming can
+be found here: (TODO add link to the main RFC).
+
+## Host and device code integration
+
+Consider this SYCL code:
+
+```
+#include <sycl.hpp>
+int main() {
+  sycl::queue q;
+  sycl::buffer<char> b{sycl::range{1024}};
+  q.submit([&](sycl::handler &cgh) {
+    sycl::accessor acc{b, cgh};
+    int i = 13;
+    struct S {
+      char c;
+      int i;
+    } test_s;
+    test_s.c = 14;
+    cgh.single_task([=] {
+      if (i == 13 && test_s.c == 14) {
+        acc[0] = 'a';
+      }
+    });
+  });
+}
+```
+
+The lambda passed to the `sycl::handler::single_task` function
+represents a device kernel. When a SYCL application is started or a
+dynamic SYCL library is loaded, all device images contained in the
+multi-targeted binary are registered in the program manager, one of the
+larger components of the SYCL runtime, using SYCL's `__tgt_register_lib`
```

Comment:
What is __tgt_register_lib? Are readers of RFC going to be aware? May be a line about it? \

Thanks

Reply:
It's LLVM's standard symbols described in https://clang.llvm.org/docs/OffloadingDesign.html, so it reasonable to expect RFC reviewers to be aware of it. It might be helpful to provide a link to the LLVM's documentation anyway.

Reply:
Added a link.